### PR TITLE
Timed events objects removal

### PR DESCRIPTION
**Pull request by dydzio0614**
Feature that was ready for some time - did small cleanup and now it looks to be ready

### CODE REVIEW

File: mapeditor/mapsettings/timedevent.h [hunk 1]
```diff
@@ -20,18 +22,27 @@ class TimedEvent : public QDialog
...
 	void on_TimedEvent_finished(int result);
 
+	void on_addObjectToDelete_clicked();
+
+	void on_removeObjectToDelete_clicked();
+
+	void onObjectPicked(const CGObjectInstance * obj);
+
+	void insertObjectToDelete(const CGObjectInstance * obj);
+
```
Comment (IvanSavenko) [inline mapeditor/mapsettings/timedevent.h:39]:
```suggestion
	void on_addObjectToDelete_clicked();
	void on_removeObjectToDelete_clicked();
	void onObjectPicked(const CGObjectInstance * obj);
	void insertObjectToDelete(const CGObjectInstance * obj);
```

File: mapeditor/mapsettings/timedevent.cpp [hunk 2]
```diff
@@ -89,10 +98,63 @@ void TimedEvent::on_TimedEvent_finished(int result)
...
+
+void TimedEvent::on_removeObjectToDelete_clicked()
+{
+	delete ui->deletedObjects->takeItem(ui->deletedObjects->currentRow());
+}
+
+void TimedEvent::onObjectPicked(const CGObjectInstance * obj)
+{
+	show();
+	dynamic_cast<QWidget *>(parent()->parent()->parent()->parent()->parent()->parent()->parent())->show();
```
Comment (IvanSavenko) [inline mapeditor/mapsettings/timedevent.cpp:136]: wtf wtf wtf wtf wtf wtf wtf is this?

File: lib/mapping/CMapDefines.h [hunk 3]
```diff
@@ -64,6 +69,14 @@ class DLL_LINKAGE CMapEvent
...
 		h & firstOccurrence;
 		h & nextOccurrence;
+		if(h.version >= Handler::Version::EVENT_OBJECTS_DELETION)
+		{
+			h & deletedObjectsInstances;
+		}
+		else
+		{
+			h & unused;
+		}
```
[the remark this answers is not in the document]
Comment (dydzio0614) [reply lib/mapping/CMapDefines.h:79]: IIRC some padding from the time when this feature was first created - related to removing another variable (and adjusting datatype to match deletedObjectsInstance is my mistake)

indeed, should not be here now, will change

File: server/processors/NewTurnProcessor.cpp [hunk 4]
```diff
@@ -41,6 +41,8 @@ NewTurnProcessor::NewTurnProcessor(CGameHandler * gameHandler)
 
 void NewTurnProcessor::handleTimeEvents(PlayerColor color)
 {
+	std::vector<const CGObjectInstance*> removedObjects;
+
```
[the remark this answers is not in the document]
Comment (dydzio0614) [reply server/processors/NewTurnProcessor.cpp:45]: will recheck

File: mapeditor/mapsettings/timedevent.cpp (hunk 2 above)
Comment (dydzio0614) [reply mapeditor/mapsettings/timedevent.cpp:136]: not my code :rofl:

Comment (IvanSavenko) [reply mapeditor/mapsettings/timedevent.cpp:136]: @godric3 then?
In any case - can this be fixed or at least explained?

Comment (godric3) [reply mapeditor/mapsettings/timedevent.cpp:136]: ah yes, guilty, copied from victory conditions (https://github.com/vcmi/vcmi/blob/develop/mapeditor/mapsettings/victoryconditions.cpp#L536)
If I remember correctly it was required to hide/show map setting dialog, to properly show map with selection layer/return to event settings. 
I don't know qt well enough to have it fixed quickly so I used same approach as found in codebase, I may look more into in on the weekend but can't promise anything

File: lib/networkPacks/NetPacksLib.cpp
Comment (Xilmi) [inline lib/networkPacks/NetPacksLib.cpp:1]: Removal of 

PlayerState * p = gs->getPlayerState(beatenHero->tempOwner);
p->removeOwnedObject(beatenHero);

Breaks the game!

**Pull request merged**